src/measurement/schema/location-schema.ts

```diff
@@ -10,42 +10,21 @@ import {regions} from '../../lib/location/regions.js';
 const {continents, countries} = geoLists;
 const measurementConfig = config.get<{limits: {global: number; location: number}}>('measurement');
 
-export const filterSchema = Joi.string().valid('combined', 'default').insensitive().default('default');
-
 export const schema = Joi.array().items(Joi.object({
-	type: Joi.string().valid('continent', 'region', 'country', 'state', 'city', 'network', 'asn', 'magic').insensitive().required(),
-	value: Joi.alternatives().conditional('type', {
-		switch: [
-			{
-				is: 'continent',
-				then: Joi.string().valid(...Object.keys(continents)).insensitive()
-					.messages({'any.only': 'The continent must be a valid two-letter ISO code'}),
-			},
-			{is: 'region', then: Joi.string().valid(...Object.keys(regions)).insensitive()},
-			{
-				is: 'country',
-				then: Joi.string().valid(...Object.keys(countries)).insensitive()
-					.messages({'any.only': 'The country must be a valid two-letter ISO code'}),
-			},
-			{
-				is: 'state',
-				then: Joi.string().valid(...Object.keys(states)).insensitive()
-					.messages({'any.only': 'The US state must be a valid two-letter code, e.g. CA'}),
-			},
-			{is: 'city', then: Joi.string().min(1).max(128).lowercase().insensitive()},
-			{is: 'network', then: Joi.string().min(1).max(128).lowercase().insensitive()},
-			{is: 'asn', then: Joi.number()},
-			{is: 'magic', then: Joi.string().min(1).insensitive()},
-		],
-	}).required().messages({
-		'any.required': 'Location value is required',
-	}),
+	continent: Joi.string().valid(...Object.keys(continents)).insensitive()
+		.messages({'any.only': 'The continent must be a valid two-letter ISO code'}),
+	region: Joi.string().valid(...Object.keys(regions)).insensitive(),
+	country: Joi.string().valid(...Object.keys(countries)).insensitive()
+		.messages({'any.only': 'The country must be a valid two-letter ISO code'}),
+	state: Joi.string().valid(...Object.keys(states)).insensitive()
+		.messages({'any.only': 'The US state must be a valid two-letter code, e.g. CA'}),
+	city: Joi.string().min(1).max(128).lowercase().insensitive(),
+	network: Joi.string().min(1).max(128).lowercase().insensitive(),
+	asn: Joi.number(),
+	magic: Joi.string().min(1).insensitive(),
 	limit: Joi.number().min(1).max(measurementConfig.limits.location).when(Joi.ref('/limit'), {
 		is: Joi.exist(),
 		then: Joi.forbidden().messages({'any.unknown': 'limit per location is not allowed when a global limit is set'}),
 		otherwise: Joi.required().messages({'any.required': 'limit per location required when no global limit is set'}),
```

Comment:
Global limit is marked as not required, because your other option is defining it per-location filter.

Reply:
Ok, that's fair, but I think it still makes sense to have sensible defaults there? Number of locations, or 1 if locations are empty.

Reply:
I think the reason why Artem decided to not set the default value for it, is the router logic. If we set it to `1` by default, we'll have hard time dinstinguishing whether user wants to use global limit, or do they want to use per-location routing - currently, we check for `globalLimit` value, and decide what to do based on that.

https://github.com/jsdelivr/globalping/blob/c30c7350d5f39bbe88b0a294f30e12b1ba3eec7b/src/probe/router.ts#L36

Reply:
We could technically loop through the `Location[]`, find an instance of an object with `limit` k/v, and assume the desired limit mechanism is location based.

Reply:
done, with tests.

Reply:
I'm missing a test for the most interesting case. Two locations and no limit.

> the limit should default to 1, or the number of requested locations if multiple are set

Reply:
Which is why I also copied how I expect it to work in that case. The default value for limit must depend on the number of locations, otherwise it doesn't make sense.

Reply:
limit = locations?.length > 1 locations.length : 1

Reply:
I don't know how would you even do that with Joi.

Reply:
But again, even if you don't set any defaults in joi and have the logic here, I'm ok with it as long as it results in the expected defaults and the only time I need to set the `limit` explicitly is when I want different value than 1 per location.

Reply:
Yeah I’m not. Joi is there for a reason - if it can’t do what we want it to do, it might be time to replace it with something else. It’s been pretty limiting throughout the way already.

Reply:
I'm fairly sure it can actually do it since, as a last resort, you can write custom validation functions.

Reply:
added custom handler.

https://github.com/jsdelivr/globalping/pull/166/commits/3fc0ea56cdc7e83ff2559e341f2bed7476416ceb